service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/CreateAzureDatabaseStep.java

```diff
@@ -58,13 +59,35 @@ private String getPodName() {
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
-    azureDatabaseUtilsRunner.createDatabaseWithDbRole(
+    var cloudContext =
         context
             .getWorkingMap()
-            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class),
-        workspaceId,
-        getPodName(),
-        resource.getDatabaseName());
+            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
+
+    // Query LZ for the postgres server
+    var bearerToken = new BearerToken(samService.getWsmServiceAccountToken());
+    var landingZoneId =
+        landingZoneApiDispatch.getLandingZoneId(
+            bearerToken, workspaceService.getWorkspace(workspaceId));
+    var databaseResource =
+        landingZoneApiDispatch
+            .getSharedDatabase(bearerToken, landingZoneId)
+            .orElseThrow(() -> new RuntimeException("No shared database found"));
+
+    // Record resource for cleanup in Janitor
+    crlService.recordAzureCleanup(
```

Comment:
Is it possible that this would be recorded twice in case of Stairway retry?

Reply:
Yes, but Janitor does handle duplicate messages. It records each event in its DB and if there is a duplicate it puts the record in DUPLICATE status and does not try to handle it. In fact I've noticed de-duplication happens fairly often for VM messages.